### PR TITLE
Problem: Support bundle hare_setup command expects conf-store URL

### PR DESCRIPTION
**Pull request by mssawant**
hare_setup support bundle invokes `hctl reportbug`, as per the `setup.yaml`
example mentioned at https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/94470861/Config+upgrade+section+in+setup.yaml,
support bundle command need not expect a configuration URL as a parameter.

Solution:
Don't add an argument to hare_setup support bundle command.

Signed-off-by: Mandar Sawant <mandar.sawant@seagate.com>

### CODE REVIEW

**Comment by mssawant** [comment]
```
[root@ssc-vm-1623:root] /opt/seagate/cortx/hare/bin/hare_setup config --config 'json:///opt/seagate/cortx/hare/conf/hare.config.conf.tmpl.3-node.sample' --file /root/config_cluster.cdf
[root@ssc-vm-1623:root]
```
```
[root@ssc-vm-1623:root] /opt/seagate/cortx/hare/bin/hare_setup init --config 'json:///opt/seagate/cortx/hare/conf/hare.config.conf.tmpl.3-node.sample' --file /root/cluster_build_330.yaml
Cluster is not running
2021-04-22 14:05:43: Generating cluster configuration... OK
2021-04-22 14:05:44: Stopping Consul on this node... OK
2021-04-22 14:05:58: Stopping Consul on other nodes... OK
2021-04-22 14:05:58: Starting Consul server on this node.............. OK
2021-04-22 14:06:10: Importing configuration into the KV store... OK
2021-04-22 14:06:10: Starting Consul on other nodes... OK
2021-04-22 14:06:11: Updating Consul configuraton from the KV store... OK
2021-04-22 14:06:12: Waiting for the RC Leader to get elected.... OK
2021-04-22 14:06:13: Starting Motr (phase1, mkfs)... OK
2021-04-22 14:06:20: Starting Motr (phase1, m0d)... OK
2021-04-22 14:06:32: Starting Motr (phase2, mkfs)... OK
2021-04-22 14:06:38: Starting Motr (phase2, m0d)... OK
2021-04-22 14:06:42: Starting S3 servers (phase3)... OK
2021-04-22 14:06:42: Checking health of services... OK
Stopping s3server@0x7200000000000001:0x17 at ssc-vm-1623.colo.seagate.com...
Stopped s3server@0x7200000000000001:0x17 at ssc-vm-1623.colo.seagate.com
Stopping m0d@0x7200000000000001:0xc (ios) at ssc-vm-1623.colo.seagate.com...
Stopped m0d@0x7200000000000001:0xc (ios) at ssc-vm-1623.colo.seagate.com
Stopping m0d@0x7200000000000001:0x9 (confd) at ssc-vm-1623.colo.seagate.com...
Stopped m0d@0x7200000000000001:0x9 (confd) at ssc-vm-1623.colo.seagate.com
Stopping hare-hax at ssc-vm-1623.colo.seagate.com...
Stopped hare-hax at ssc-vm-1623.colo.seagate.com
Cluster is not running
[root@ssc-vm-1623:root]
```
```
[root@ssc-vm-1623:root] /opt/seagate/cortx/hare/bin/hare_setup test --config 'json:///opt/seagate/cortx/hare/conf/hare.config.conf.tmpl.3-node.sample' --file /root/cluster_build_330.yaml
Cluster is not running
2021-04-22 14:07:30: Generating cluster configuration... OK
2021-04-22 14:07:32: Stopping Consul on this node... OK
2021-04-22 14:07:43: Stopping Consul on other nodes... OK
2021-04-22 14:07:43: Starting Consul server on this node......... OK
2021-04-22 14:07:50: Importing configuration into the KV store... OK
2021-04-22 14:07:50: Starting Consul on other nodes... OK
2021-04-22 14:07:50: Updating Consul configuraton from the KV store... OK
2021-04-22 14:07:51: Waiting for the RC Leader to get elected......... OK
2021-04-22 14:07:58: Starting Motr (phase1, m0d)... OK
2021-04-22 14:08:02: Starting Motr (phase2, m0d)... OK
2021-04-22 14:08:05: Starting S3 servers (phase3)... OK
2021-04-22 14:08:06: Checking health of services... OK
Stopping s3server@0x7200000000000001:0x17 at ssc-vm-1623.colo.seagate.com...
Stopped s3server@0x7200000000000001:0x17 at ssc-vm-1623.colo.seagate.com
Stopping m0d@0x7200000000000001:0xc (ios) at ssc-vm-1623.colo.seagate.com...
Stopped m0d@0x7200000000000001:0xc (ios) at ssc-vm-1623.colo.seagate.com
Stopping m0d@0x7200000000000001:0x9 (confd) at ssc-vm-1623.colo.seagate.com...
Stopped m0d@0x7200000000000001:0x9 (confd) at ssc-vm-1623.colo.seagate.com
Stopping hare-hax at ssc-vm-1623.colo.seagate.com...
Stopped hare-hax at ssc-vm-1623.colo.seagate.com
Cluster is not running
```
```
[root@ssc-vm-1623:root] /opt/seagate/cortx/hare/bin/hare_setup support_bundle
Cluster is not running
Created /tmp/hare/hare_ssc-vm-1623.tar.gz
[root@ssc-vm-1623:root]
```
```
[root@ssc-vm-1623:root] /opt/seagate/cortx/hare/bin/hare_setup prepare --config 'json:///opt/seagate/cortx/hare/conf/hare.config.conf.tmpl.3-node.sample'
[root@ssc-vm-1623:root]
```

**Comment by SwapnilGaonkar7** [comment]
@mssawant Sorry if it is already discussed while implementing these changes but isn't '--config $URL' mandatory for all the stages present in setup.yaml.
Also do you think we should get it reviewed by someone from Provisioner team?

**Comment by knekrasov** [comment]
@SwapnilGaonkar7 
> Sorry if it is already discussed while implementing these changes but isn't '--config $URL' mandatory for all the stages present in setup.yaml.
Also do you think we should get it reviewed by someone from Provisioner team?

That requirement comes from some weakness of Mini Provisioner's parser. I would propose a bit different approach: make `--config` argument optional (so that if Mini Provisioner invokes it with this argument, no fail will happen) but do not require it. In that case the end users are not forced to provide `--config` with ConfStore file which is effectively meaningless. I think it is way more user friendly.

**Comment by knekrasov** [comment]
@mssawant I took a liberty to add a fixup commit that makes `--config` optional for support_bundle, cleanup and reset (so if it is provided, it will not break anything) and polished the argparse stuff a bit. If you like the change, please squash it into your PR.

**Review by knekrasov — APPROVED** [review]

File: provisioning/setup.yaml [hunk 1]
```diff
@@ -23,3 +20,5 @@ hare:
   prepare:
     cmd: /opt/seagate/cortx/hare/bin/hare_setup prepare
     args: --config $URL
+support_bundle:
```
Comment (suvratjoshi) [inline provisioning/setup.yaml:23]: Looks here too needs some indentations correcion

**Comment by suvratjoshi** [comment]
Single node, three-node wikis and RFC - 20/MINIPROV also need to be updated with these argument details I guess

File: provisioning/miniprov/hare_mp/main.py [hunk 2]
```diff
@@ -350,38 +354,43 @@ def main():
...
-                   help_str='Reset/cleanup step',
-                   handler_fn=noop)
-    add_subcommand(subparser,
-                   'cleanup',
-                   help_str='Reset/cleanup step',
-                   handler_fn=noop)
+                   help_str='Resets temporary Hare data and configuration',
+                   handler_fn=noop, config_required=False)
+
+    add_subcommand(
```
[the remark this answers is not in the document]
Comment (knekrasov) [reply provisioning/miniprov/hare_mp/main.py:383]: No, I believe they are fine.

**Comment by mssawant** [comment]
@SwapnilGaonkar7 , if you look at the setup.yaml mentioned at https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/94470861/Config+upgrade+section+in+setup.yaml and also if you check it for other components support bundle generation command does not take any argument. Also the hierarchy in Hare setup.yaml is not what is expected as mentioned in https://jts.seagate.com/browse/EOS-19522?focusedCommentId=2089987&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2089987

File: provisioning/setup.yaml (hunk 1 above)
Comment (mssawant) [reply provisioning/setup.yaml:23]: No, indentation is correct, support_bundle key is not expected to be a child of hare key as per the setup.yaml template mentioned in the commit message.

**Comment by mssawant** [comment]
@knekrasov, thanks for the commit.

**Review by SwapnilGaonkar7 — APPROVED** [review]

**Pull request merged**